### PR TITLE
Enable markdown for description in data-container

### PR DESCRIPTION
**Pull request by XiaoxuHU**
image

- Enable markdown for description in data-container
- Create a new Component called Markdown-Text, to refactor and remove some duplicate code in Insight component and Markdown-View component

### CODE REVIEW

File: AngularApp/projects/diagnostic-data/src/lib/components/data-container/data-container.component.html [hunk 1]
```diff
@@ -15,7 +15,10 @@
       </ng-container>
     </div>
     <div *ngIf="description" class="panel-heading text-muted card-description" [class.with-top-border]="!title">
-      {{description}}
+      <!-- {{description}} -->
+      <!-- <div [hidden]="isMarkdown(description)" [innerHtml]="description"></div>
+      <markdown #markdownDiv [data]="getMarkdown(description)" [hidden]="!isMarkdown(description)"></markdown> -->
```
Comment (puneetg1983) [inline AngularApp/projects/diagnostic-data/src/lib/components/data-container/data-container.component.html:20]: Want to remove the commented out portion?

File: AngularApp/projects/diagnostic-data/src/lib/components/markdown-text/markdown-text.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  getMarkdown(str: string) {
+    return str.trim().replace('<markdown>', '').replace('</markdown>', '');
+  }
+
+  ngAfterViewInit() {
+    if (this.markdownDiv) {
+      this.listenObj = this.renderer.listen(this.markdownDiv.element.nativeElement, 'click', (evt) => {
+        this.linkInterceptorService.interceptLinkClick(evt, this.router, this.detector, this.telemetryService)
```
Comment (puneetg1983) [inline AngularApp/projects/diagnostic-data/src/lib/components/markdown-text/markdown-text.component.ts:36]: >this.linkInterceptorService.interceptLinkClick [](start = 8, length = 46)

Can you please make sure that this function is getting clicked for all the hyperlinks after the refactoring of the code ?

File: AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html [hunk 3]
```diff
@@ -11,14 +11,14 @@
...
+    <markdown-text [markdownData]="markdownData" [detector]="detector" [isFromMarkdownView]="false"></markdown-text>
 </data-container>
 
 <!-- If there is no title, do not use container. Note this is called from the else statement in the *ngIf above -->
 <ng-template #noContainerMarkdown>
     <div *ngIf="renderingProperties.title">
         <h1>{{renderingProperties.title}}</h1>
     </div>
-    <markdown #markdownDiv [data]="markdownData"></markdown>
+    <markdown-text [markdownData]="markdownData" [isFromMarkdownView]="false"></markdown-text>
```
Comment (puneetg1983) [inline AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html:22]: > [isFromMarkdownView]="false" [](start = 48, length = 29)

The property name is slightly confusing, We are in the markdown-view.component but isFromMarkdownView is false?

File: AngularApp/projects/diagnostic-data/src/lib/components/data-container/data-container.component.html (hunk 1 above)
Comment (XiaoxuHU) [reply AngularApp/projects/diagnostic-data/src/lib/components/data-container/data-container.component.html:20]: Yes, will do

File: AngularApp/projects/diagnostic-data/src/lib/components/markdown-text/markdown-text.component.ts (hunk 2 above)
Comment (XiaoxuHU) [reply AngularApp/projects/diagnostic-data/src/lib/components/markdown-text/markdown-text.component.ts:36]: Yes, Puneet. I tested for external link and internal link inside markdown. Both should be works fine.

File: AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html (hunk 3 above)
Comment (XiaoxuHU) [reply AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html]: Sorry, my fault. I updated the code so it passed as "true" now.

Comment (puneetg1983) [reply AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html]: The isForMarkdownView still sounds a bit wierd. Maybe use `isMarkdownView` or `fromMarkdownview`?

Comment (XiaoxuHU) [reply AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.html]: Yes, let me change the property name. Thank you, Puneet !

**Pull request merged**